src/Illuminate/Routing/Router.php

```diff
@@ -454,12 +462,39 @@ public function getLastGroupPrefix()
      * @param  string  $uri
      * @param  array|string|callable|null  $action
      * @return \Illuminate\Routing\Route
+     * @throws \Illuminate\Routing\Exceptions\RouteAlreadyRegisteredException
      */
     public function addRoute($methods, $uri, $action)
     {
+        $this->validateDuplicatesIfNeeded($methods, $uri);
+
         return $this->routes->add($this->createRoute($methods, $uri, $action));
     }
 
+    /**
+     * Throw an exception if a route with the same path already exists.
+     *
+     * @param  array|string  $methods
+     * @param  string  $uri
+     * @return void
+     * @throws \Illuminate\Routing\Exceptions\RouteAlreadyRegisteredException
+     */
+    protected function validateDuplicatesIfNeeded($methods, $uri)
```

Comment:
```suggestion
    protected function validateDuplicatesIfNeeded($newRoute)
```

How about passing the new Route, which should be checked against the other routes?
You would then have access to check the domain, which should solve your problem with the failing test.

Reply:
You're right... I tried doing that but only after adding the route, so it obviously didn't work. 😅 thank you